samples/bluetooth/direct_test_mode/src/fem/nrf21540.c

```diff
@@ -434,14 +460,38 @@ int nrf21540_init(void)
 
 	k_busy_wait(NRF21540_PD_PG_TIME_US);
 
+	atomic_set(&state, NRF21540_STATE_READY);
+
+	return 0;
+}
+
+int nrf21540_powerdown(void)
+{
+	int err;
+
+	if (!atomic_cas(&state, NRF21540_STATE_READY, NRF21540_STATE_BUSY)) {
+		return -EBUSY;
+	}
+
+	/* Power-down nRF21540 */
+	err = gpio_pin_set(nrf21540_cfg.pdn.port, nrf21540_cfg.pdn.pin, 0);
+	if (err) {
+		return err;
+	}
+
+	atomic_set(&state, NRF21540_STATE_OFF);
+
 	return 0;
```

Comment:
What about the _PG->PD settling time_, shouldn't the code wait here? Or can we safely assure the PDN won't be activated before this time?

Reply:
I think it isn't necessary because we don't have a use case where we power-up the nrf21540 immediately after power-down. I think code execution takes more than settling time. I can write in this function documentation that the settling time between those states is 10us and this function doesn't wait.

Reply:
That's correct, no need to add a busy wait here.